lib/cc/analyzer/include_paths_builder.rb

```diff
@@ -0,0 +1,150 @@
+module CC
+  module Analyzer
+    class IncludePathsBuilder
+      def self.relevant_entries(path)
+        Dir.entries(path).reject do |e|
+          %w(. .. .git).include?(e)
+        end
+      end
+
+      def initialize(cc_exclude_paths)
+        @cc_exclude_paths = cc_exclude_paths
+      end
+
+      def build
+        root = Directory.new('.', ignored_files)
+        paths = root.included_paths
+        paths.each do |path|
+          raise_on_unreadable_files(path)
+        end
+      end
+
+      protected
+
+      def ignored_files
+        Tempfile.open(".cc_gitignore") do |tmp|
+          tmp.write(File.read(".gitignore")) if File.exist?(".gitignore")
+          tmp << @cc_exclude_paths.join("\n")
+          tmp.close
+          tracked_and_ignored = `git ls-files -zi -X #{tmp.path}`.split("\0")
+          untracked_and_ignored = `git ls-files -zio -X #{tmp.path}`.split("\0")
+          tracked_and_ignored + untracked_and_ignored
+        end
+      end
+
+      def raise_on_unreadable_files(path)
+        if File.directory?(path)
```

Comment:
At what point are you actually looking to see if directories themselves are "unreadable"?